lib/handleRequest.js

```diff
@@ -125,28 +129,64 @@ function validationCompleted (request, reply, validationErr) {
 function preHandlerCallback (err, request, reply) {
   if (reply.sent) return
 
-  if (err != null) {
-    reply[kReplyIsError] = true
-    reply.send(err)
-    return
+  const context = request[kRouteContext]
+
+  const store = {
+    request,
+    reply,
+    async: false,
+    route: {
+      url: context.config.url,
+      method: context.config.method
+    }
```

Comment:
Can we avoid allocating those two objects if there are no diagnostics channels?

Reply:
The creation of the store object is now wrapped in a check to see if there are any listeners on any of the channels.